csharp_dotnetcore/18.Multi-Lingual-Bot/Translation/MicrosoftTranslator.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using MultiLingualBot.Translation.Model;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MultiLingualBot.Translation
+{
+    public interface ITranslator
+    {
+        Task<string> TranslateAsync(string text, string targetLocale);
+    }
+
+    public class MicrosoftTranslator : ITranslator
+    {
+        private const string host = "https://api.cognitive.microsofttranslator.com";
+        private const string path = "/translate?api-version=3.0";
+        private const string uriParams = "&to=";
+
+        private readonly string _key = "ENTER KEY HERE";
+
+        public MicrosoftTranslator(string key)
+        {
+            _key = key;
+        }
+
+        public async Task<string> TranslateAsync(string text, string targetLocale)
+        {
+            // From Cognitive Services translation documentation:
+            // https://docs.microsoft.com/en-us/azure/cognitive-services/translator/quickstart-csharp-translate
+            var body = new object[] { new { Text = text } };
+            var requestBody = JsonConvert.SerializeObject(body);
+
+            using (var client = new HttpClient())
+            using (var request = new HttpRequestMessage())
```

Comment:
The HttpClient needs to be static on the class. If you don't do that, every single turn will create a new HTTPS connection, which means latency will be like 800ms per turn just in HTTPS setup time.